### ISSUE TITLE
the centroid calculation is wrong

### ISSUE THREAD

**Issue by hdahmani1** [tmgerard/Matrix-Structural-Analysis-Excel#4]
I noticed that the formula for calculating centroid of polygon is wrong. What you've calculated is the average of points NOT centroid.
the formula of centroid is : 
![image](https://user-images.githubusercontent.com/76886002/199461932-bab70796-9af9-4e2d-9d95-a312a68e5072.png)

**Comment by tmgerard** [comment]
Thanks! I goofed on that one. The equation you are showing looks like it is for a collection of regular shapes, which might not be the best fit for a polygon. I found another formula that looks like it was derived similarly to the shoelace algorithm I used for the area calculation that might be a better fit.

[https://en.wikipedia.org/wiki/Polygon#Centroid](https://en.wikipedia.org/wiki/Polygon#Centroid)

**Comment by tmgerard** [comment]
I added the area method and updated the centroid method. I thought I already had an area method, but that was something I did in another project.  I added a few extra tests to check the area and centroid calculations for when the points of the polygon are added in a clockwise and counterclockwise order, which will affect the sign of the area.

[Changes to Polygon2D class](https://github.com/tmgerard/Matrix-Structural-Analysis-Excel/commit/e279e24c163cc2f22788c4b1a21e42aeca356204)
